.github/workflows/ci.yml

```diff
@@ -49,71 +40,14 @@ jobs:
   docs:
     name: Documentation
     runs-on: ubuntu-latest
-    env:
-      JULIA_PKG_SERVER: ""
     steps:
       - uses: actions/checkout@v2
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
       - uses: julia-actions/setup-julia@v1
         with:
           version: '1'
-      - run: |
-         julia -e '
-           function set_environment_variable(name::AbstractString, value::AbstractString)
-               github_env = ENV["GITHUB_ENV"]
-               touch(github_env)
-               open(github_env, "a") do io
-                   println(io, "$(name)=$(value)")
-               end
-           end
-           event_name = "${{ github.event_name }}"
-           if event_name == "pull_request"
-               base_ref = "${{ github.base_ref }}"
-               head_ref = "${{ github.head_ref }}"
-               base_repository = "${{ github.repository }}"
-               head_repository = "${{ github.event.pull_request.head.repo.full_name }}"
-               build_docs = (base_ref == "master") && (head_ref == "dev") && (base_repository == head_repository)
-           elseif event_name == "push"
-               ref = "${{ github.ref }}"
-               build_docs = (ref == "refs/heads/master") || (startswith(ref, "refs/tags/"))
-           elseif event_name == "schedule"
-               build_docs = ref == "refs/heads/master"
-           elseif event_name == "workflow_dispatch"
-               build_docs = ref == "refs/heads/master"
-           else
-               build_docs = false
-           end
-           if build_docs
-               @info("We will build the docs")
-               set_environment_variable("BUILD_DOCS", "true")
-           else 
-               @info("We will NOT build the docs")
-               set_environment_variable("BUILD_DOCS", "false")
-           end'
-      - run: |
-          julia --project=docs -e '
-            if ENV["BUILD_DOCS"] == "true"
-                using Pkg
-                Pkg.develop(PackageSpec(path=pwd()))
-                Pkg.instantiate()
-            end'
-      - run: |
-          julia --project=docs -e '
-            if ENV["BUILD_DOCS"] == "true"
-                using Documenter: doctest
-                using MLJ
-                @info "attempting to run the doctests"
-                doctest(MLJ)
-            else
-                @info "skipping the doctests"
-            end'
-      - run: julia --project=docs -e '
-            if ENV["BUILD_DOCS"] == "true"
-                @info "attempting to build the docs"
-                run(`julia --project=docs docs/make.jl`)
-                @info "successfully built the docs"
-            else
-                @info "skipping the docs build"
-            end'
+      - uses: julia-actions/julia-docdeploy@v1
```

Comment:
The logic of the part above this appears to be about skipping builds in certain situations. I think it's safe to switch to `julia-docdeploy` since that is more idiomatic and simpler

Reply:
Yes, we only want to generate documentation on dev -> master PR's because it takes a very long time to run. Is that preserved in this PR?

Reply:
No that wasn't preserved. I can put it back. Just to double check, it was meant to save computation from GitHub? The current approach takes a bit shorter to run because it does the doctests and documentation in one step, saving some re-compilation time. Also, the running time appears to be 15 minutes for the documentation. This seems reasonable as `Turing.jl` runs 6 times 60 minutes and `LoopVectorization.jl` runs 20 or so jobs, some of which taking hours.

Reply:
Yes, but those in those packages - I'm guessing - PR's are made directly onto master, rather than first onto dev. 
I'd prefer to suppress the doc generation for "staging" PR's with dev as target. But I also appreciate the complexity and likely brittleness of the logic there. 

@DilumAluthge Any thoughts?

Reply:
I've put the old config back in the commits below (everything after 870b30e).